### PR TITLE
[OpenAPI] update 2024-02-01 to 2024-06-01

### PR DESCRIPTION
**Pull request by newtork**
## Context

https://github.tools.sap/AI/ai-sdk-java-backlog/issues/118

Specs can be compared manually here:
* https://github.com/Azure/azure-rest-api-specs/tree/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/stable

## Definition of Done

- [x] Functionality scope stated & covered
- [ ] Tests cover the scope above
  - [x] E2E tested
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Aligned changes with the JavaScript SDK~
- [x] ~Documentation updated~
- [x] ~Release notes updated~

### CODE REVIEW

File: foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/model/OpenAiEmbeddingParameters.java [hunk 1]
```diff
@@ -40,6 +40,21 @@ public class OpenAiEmbeddingParameters {
...
+  @Setter(onParam_ = @Nullable)
+  private String encodingFormat;
+
+  /**
+   * The number of dimensions the resulting output embeddings should have. Only supported in
+   * `text-embedding-3` and later models.
+   */
+  @JsonProperty("dimensions")
+  @Setter(onParam_ = @Nullable)
+  private Integer dimensions;
```
Comment (newtork) [inline foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/model/OpenAiEmbeddingParameters.java:56]: (Major)

> The number of dimensions the resulting output embeddings should have.

The only text embedding model available for testing is `TEXT_EMBEDDING_ADA_002`.
Unfortunately it does not change output upon setting this field. No matter `null`, `0`, `1`, or `5`.

> Only supported in `text-embedding-3` and later models.

Should we release something that we cannot test?

File: foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/model/OpenAiCompletionParameters.java [hunk 2]
```diff
@@ -109,6 +109,24 @@ public class OpenAiCompletionParameters {
...
   private Boolean stream;
 
+  /**
+   * Whether to return log probabilities of the output tokens or not. If true, returns the log
+   * probabilities of each output token returned in the `content` of `message`. This option is
+   * currently not available on the `gpt-4-vision-preview` model. Default: false.
+   */
+  @JsonProperty("logprobs")
+  @Setter(onParam_ = @Nullable)
+  private Boolean logprobs;
```
[the remark this answers is not in the document]
Comment (newtork) [reply foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/model/OpenAiCompletionParameters.java:119]: This is what it looks like :+1:

image

**Review by CharlesDuboisSAP — COMMENTED** [review]
You should test what you add, otherwise what's the point of adding it since none of these additions are required anyway?

**Comment by newtork** [comment]
We decided not to merge it for GA.

**Pull request closed without merging**